rust/rsc/src/bin/rsc/read_job.rs

```diff
@@ -34,26 +35,51 @@ async fn record_miss(hash: String, conn: Arc<DatabaseConnection>) {
 }
 
 #[tracing::instrument(skip(db, stores))]
-async fn resolve_blob(
-    id: Uuid,
-    db: &DatabaseTransaction,
+async fn resolve_blobs<T: ConnectionTrait>(
+    ids: &[Uuid],
+    db: &T,
     stores: &HashMap<Uuid, Arc<dyn blob::DebugBlobStore + Sync + Send>>,
-) -> Result<ResolvedBlob, String> {
-    let Ok(Some(blob)) = entity::prelude::Blob::find_by_id(id).one(db).await else {
-        return Err(format!("Unable to find blob {} by id", id));
-    };
+) -> Result<HashMap<Uuid, ResolvedBlob>, String> {
+    // Fetch all blobs in a single query
+    let blobs = Blob::find()
+        .filter(entity::blob::Column::Id.is_in(ids.to_vec()))
+        .all(db)
+        .await
+        .map_err(|e| format!("Failed to query blobs: {}", e))?;
 
-    let Some(store) = stores.get(&blob.store_id) else {
-        return Err(format!(
-            "Unable to find backing store {} for blob {}",
-            blob.store_id, id
-        ));
-    };
+    // Build a map of blob_id -> blob model for quick lookup
+    let blob_map: HashMap<Uuid, entity::blob::Model> = blobs.into_iter().map(|b| (b.id, b)).collect();
+
+    // Ensure we have all requested blobs
+    for &id in ids {
+        if !blob_map.contains_key(&id) {
+            return Err(format!("Unable to find blob {} by id", id));
+        }
+    }
 
-    return Ok(ResolvedBlob {
-        id: blob.id,
-        url: store.download_url(blob.key).await,
+    // Resolve all download URLs in parallel
+    let futures = blob_map.iter().map(|(id, blob)| {
+        let store_opt = stores.get(&blob.store_id).cloned();
+        let key = blob.key.clone();
+
+        async move {
+            let store = store_opt.ok_or_else(|| {
+                format!("Unable to find backing store {} for blob {}", blob.store_id, id)
+            })?;
+            let url = store.download_url(key).await;
+            Ok::<(Uuid, ResolvedBlob), String>((*id, ResolvedBlob { id: *id, url }))
+        }
     });
+
+    let results = join_all(futures).await;
+
+    let mut resolved_map = HashMap::new();
+    for res in results {
+        let (id, resolved_blob) = res?;
+        resolved_map.insert(id, resolved_blob);
+    }
```

Comment:
Is there no simpler constructor for this?
We only to do this iteratively because we wanted to resolve the urls in parallel, rather than with a map on the original `blob_map`?